apps/site/components/EOL/Modal.tsx

```diff
@@ -0,0 +1,156 @@
+import { Modal, Title, Content } from '@node-core/ui-components/Common/Modal';
+import classNames from 'classnames';
+import { useTranslations } from 'next-intl';
+import type { FC } from 'react';
+
+import VulnerabilityChip from '#site/components/EOL/VulnerabilityChips/Chip';
+import LinkWithArrow from '#site/components/LinkWithArrow';
+import type { ModalProps } from '#site/providers/modalProvider';
+import type { NodeRelease } from '#site/types';
+import type { Vulnerability } from '#site/types/vulnerabilities';
+
+import { SEVERITY_ORDER } from './VulnerabilityChips';
+
+type EOLModalData = {
+  release: NodeRelease;
+  vulnerabilities: Array<Vulnerability>;
+};
+
+type KnownVulnerability = Vulnerability & {
+  severity: (typeof SEVERITY_ORDER)[number];
+};
+
+const VulnerabilitiesTable: FC<{
+  vulnerabilities: Array<Vulnerability>;
+  maxWidth?: string;
+}> = ({ vulnerabilities, maxWidth = 'max-w-2xs' }) => {
+  const t = useTranslations('components.eolModal');
+
+  return (
+    <table className="w-full">
+      <thead>
+        <tr>
+          <th>{t('table.cves')}</th>
+          <th>{t('table.severity')}</th>
+          <th>{t('table.overview')}</th>
+          <th>{t('table.details')}</th>
+        </tr>
+      </thead>
+      <tbody>
+        {vulnerabilities.map((vuln, i) => (
+          <tr key={i}>
+            <td>
+              {vuln.cve.length
+                ? vuln.cve.map(cveId => (
+                    <div key={cveId}>
+                      <LinkWithArrow
+                        href={`https://www.cve.org/CVERecord?id=${cveId}`}
+                        target="_blank"
+                        rel="noopener noreferrer"
+                      >
+                        {cveId}
+                      </LinkWithArrow>
+                    </div>
+                  ))
+                : '-'}
+            </td>
+            <td>
+              <VulnerabilityChip severity={vuln.severity} />
+            </td>
+            <td className={classNames(maxWidth, 'truncate')}>
+              {vuln.description || vuln.overview || '-'}
+            </td>
+            <td>
+              {vuln.ref ? (
+                <LinkWithArrow
+                  href={vuln.ref}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  {t('blogLinkText')}
+                </LinkWithArrow>
+              ) : (
+                '—'
+              )}
+            </td>
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+const UnknownSeveritySection: FC<{
+  vulnerabilities: Array<Vulnerability>;
```

Comment:
This implies that UnknownSeveritySection will allow empty values.

I'd also argue the logic for rendering unknown vulnerabilities should be responsiiblity of this component. So you pass all vulnerabilities to this component and it decides whaty to render.